### PR TITLE
Implement GetGroup api endpoint

### PR DESCRIPTION
**Pull request by pkulik0**
## Description

This PR adds a GetGroup (by uuid) facade to JIMM to enable easy retrieval of groups in the juju terraform provider.

Fixes [CSS-10682](https://warthogs.atlassian.net/browse/CSS-10682)

## Engineering checklist

- [ ] Documentation updated
- [x] Covered by unit tests
- [ ] Covered by integration tests

[CSS-10682]: https://warthogs.atlassian.net/browse/CSS-10682?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by kian99 — DISMISSED** [review]
LGTM nice stuff! size S

File: internal/jujuapi/access_control.go [hunk 1]
```diff
@@ -58,6 +58,23 @@ func (r *controllerRoot) AddGroup(ctx context.Context, req apiparams.AddGroupReq
 	return resp, nil
 }
 
+// GetGroup retrieves a group within JIMMs DB for reference by OpenFGA.
```
Comment (kian99) [inline internal/jujuapi/access_control.go]: This docstring could be simpler 
```suggestion
// GetGroup returns group information based on a group ID.
```

Comment (pkulik0) [reply internal/jujuapi/access_control.go]: Other methods have godocs like this so I just went with the convention but didn't like it too. We can just simplify all of them then.

**Review by kian99 — APPROVED** [review]

**Review by SimoneDutto — APPROVED** [review]
lgtm

size: S

**Pull request merged**